TESTS/netsocket/tcp/tcpsocket_bind_address.cpp

```diff
@@ -39,7 +39,12 @@ void TCPSOCKET_BIND_ADDRESS()
     }
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock->open(get_interface()));
     SocketAddress sockAddr = SocketAddress(get_interface()->get_ip_address(), 80);
-    TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock->bind(sockAddr));
+    nsapi_error_t bind_result = sock->bind(sockAddr);
+    if (bind_result == NSAPI_ERROR_UNSUPPORTED) {
+        TEST_IGNORE_MESSAGE("bind() not supported");
```

Comment:
Does the `TEST_IGNORE_MESSAGE` return from this function?

Or do we need to have `return` after each ignore case?

Reply:
I haven't found any proper documentation of Unity, but looking at the code, the [exit is only called ](https://github.com/ARMmbed/mbed-os/blob/ac361d6a4bc2675285436ce8ffae955c2abe0c71/features/frameworks/utest/source/utest_harness.cpp#L209)if `fail_status == STATUS_ABORT ` . The [default greentea assertion handler ](https://github.com/ARMmbed/mbed-os/blob/ac361d6a4bc2675285436ce8ffae955c2abe0c71/features/frameworks/utest/source/utest_default_handlers.cpp#L102), in case of the reason for failure being IGNORE, we will set the fail_status to STATUS_IGNORE.
This also makes sense, as we need to clean up the dynamically allocated Socket...